### ISSUE TITLE
Makefile fails to find llvm-config

### ISSUE THREAD

**Issue by ikirill** [ihnorton/Clang.jl#114]
Makefile expects to find llvm-config under julia's root dir, but on my machine I have llvm-config in PATH from a recent separate llvm install, and there's nothing anywhere in julia's install dir. This is with Julia-0.3.4 on OSX.

The offending line seems to be this one:
https://github.com/ihnorton/Clang.jl/blob/0b760af56ad10670edca500c6fab4df0a3cc9581/deps/src/Makefile#L20

leading to compilation errors in Pkg.add:

     make: /opt/software/Julia-0.3.4.app/Contents/Resources//usr/bin/llvm-config: Command not found
     g++ wrapclang.cpp.o -rdynamic -shared -o libwrapclang.dylib -L   -lclang
     ld: warning: directory not found for option '-L-lclang'
     Undefined symbols for architecture x86_64:
       "_clang_CXXMethod_isStatic", referenced from:
           _wci_CXXMethod_isStatic in wrapclang.cpp.o

**Comment by ihnorton** [comment]
@staticfloat is there any way to get the Julia build Makefile options from the Homebrew or PPA versions? In particular, to check whether `USE_SYSTEM_LLVM` is set?

**Comment by staticfloat** [comment]
Hmmm.  No, I don't think so, since by the time we get to runtime everything is gonna look pretty similar.

Binary distributions aren't going to ship with `LLVM` at all though, so I think the best answer here right now is to check for `llvm-config` and if it isn't in any of the usual places, (on the `PATH`, in `$(JULIA_ROOT)/usr/bin/llvm-config`) require the user to install it.

**Comment by ikirill** [comment]
The problem with the Makefile right now is that it doesn't check path. I had to modify the Makefile myself. (But not the julia version from git.)

**Comment by staticfloat** [comment]
Right, what I'm suggesting is that it first checks `$(JULIA_ROOT)/usr/bin` for `llvm-config`, and if it doesn't find that, it checks for `PATH` to see if it contains `llvm-config`, and if not that, it exits out angrily.

**Comment by ikirill** [comment]
Also, is there a reason why LLVM version is hardcoded in the Makefile as 3.3?

**Comment by ihnorton** [comment]
I've reworked the Makefile to search for llvm-config, and removed hard-coded versions. Unfortunately I haven't been able to get Travis work yet because the nightly binaries use their own build of LLVM which seems to conflict with the system libclang/libLLVM. At least that is my guess - the helper library does build and tests do pass using the nightlies on my local computer so I'm not entirely sure.

**Comment by ScottPJones** [comment]
This seems to be what is breaking for me, on a Ubuntu 14.04 VM, 64-bit, with release 0.4.0.
```
INFO: Building Clang
========================================================[ ERROR: Clang ]========================================================

LoadError: failed process: Process(`which llvm-config`, ProcessExited(1)) [1]
while loading /home/vagrant/.julia/v0.4/Clang/deps/build.jl, in expression starting on line 10

================================================================================================================================
```
There is a `llvm-config` present, but it is `llvm-config-3.7`, not plain `llvm-config`

If I make a link, so that it can find `llvm-config`, then it failed because the version with fixes for 0.4
is not tagged (i.e. DL_LOAD_PATH not defined error comes up).
If I correct that, then I run into the following error:
```
julia> Pkg.build("Clang")
INFO: Building Clang
g++ wrapclang.cpp -fPIC -c -o wrapclang.cpp.o -I/usr/lib/llvm-3.7/include -g -Wall -Wno-strict-aliasing -fno-omit-frame-pointer -fPIC 
wrapclang.cpp:7:27: fatal error: clang-c/Index.h: No such file or directory
 #include "clang-c/Index.h"
                           ^
compilation terminated.
make: *** [wrapclang.cpp.o] Error 1
```
Where should it be getting that from?

Thanks, Scott

**Comment by LordZorgoth** [comment]
It's basically the same issue I think. Mine is in "/usr/lib/llvm-3.6/include/clang-c/Index.h"

Rather than making a link, I simply altered the line that looked for llvm-config.

</code>tmp=joinpath(JULIA_HOME,readdlm(IOBuffer(readall(pipeline(`ls $JULIA_HOME`,`grep "^llvm-config.*"`))))[end])</code>

Perhaps the Makefile should be updated in this way? The behavior Scott described will happen to anyone who downloads llvm from the Ubuntu repositories.
